code/modules/events/zombie_outbreak.dm

```diff
@@ -0,0 +1,29 @@
+/datum/round_event_control/zombie_outbreak
+	name = "Zombie Outbreak"
+	typepath = /datum/round_event/zombie_outbreak
+	weight = 5
+	max_occurrences = 1
+	min_players = 30
+
+/datum/round_event/zombie_outbreak
+	fakeable = FALSE
+
+/datum/round_event/zombie_outbreak/start()
+	// The amount of infections among the crew depends on the amount of living, non-AFK, non-silicon players on the station
+	var/list/living_crew = list()
+	living_crew = get_living_station_crew()
+	var/infection_count = round(length(living_crew) / 10)
+
+	for(var/i = 0, i < infection_count, i++)
+		for(var/mob/living/carbon/human/H in shuffle(GLOB.player_list))
+			if(H.stat == DEAD)
+				continue
+			if(!SSjob.GetJob(H.mind.assigned_role) || (H.mind.assigned_role in GLOB.nonhuman_positions))
+				continue
+			if(!H.getorgan(/obj/item/organ/brain))
+				continue
+			if(!H.getorganslot(ORGAN_SLOT_ZOMBIE))
+				var/obj/item/organ/zombie_infection/ZI = new()
+				ZI.Insert(H)
```

Comment:
Needs to make sure they aren't immune to zombie infections (IPCs and plasmamen are I believe).
I think it would be better if this was some kind of thing that spawns in maints and can infect someone rather than just instantly giving someone a zombie infection

Reply:
fixed. also, the idea is that it's a race against time to see if medical can get rid of the infections before they actually turn victims into zombies. however, since it scales with station population up to 5 concurrent infections, there is a high likelihood that at least someone gets turned into a zombie and starts spreading.

also having a zombie spawn in maints would both be less dramatic and also make a zombie outbreak way harder to get off the ground than 3-5 random infections on the station.

Reply:
I still think a good alternative for the outbreak would be to make the patient zeros to be sleeping syndicate agents with the objective to spread it, you could reduce the sleeper agents to be a maximum of three scaling with pop, i suggest that because i just feel it will be fine at first but people are gonna start meta-checking for tumors the moment a body with no harmful reagents and no radiation starts died of toxins 
ideally you could even make them to not be taking slow toxi damage until death but rather a transformation(like if they used the magic mirror) 

Buuut now that i think about it.. in this current iteration you coded does the player gets a notification or they'll just start taking toxin damage? 

I vaguely remember a message that you get when you first die to that toxin damage like "you're dead but you'll rise!" Maybe check if the font size is big enough? At the very least they can get mentally prepared for the moment they revive they're gonna be zombo

I do agree the worst implementation would be the zombie mob since it would get META'd fast

Reply:
you simply grow a romerol tumor. you do get a message in chat about zombification when you die, with big red text.